core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -578,8 +578,10 @@ private RexNode simplifyCoalesce(RexCall call) {
     final Set<String> digests = new HashSet<>();
     final List<RexNode> operands = new ArrayList<>();
     for (RexNode operand : call.getOperands()) {
-      operand = simplify_(operand);
-      if (digests.add(operand.digest)) {
+      operand = withUnknownAsFalse(false).simplify_(operand);
+      if (digests.add(operand.digest)
+          && (!RexUtil.isNull(operand)
+                  || !unknownAsFalse)) {
```

Comment:
RexUtil.isNull and unknownAsFalse look very odd here

Reply:
well,In unknownAsFalse mode,Coalesce(null,x) will simplify to x